devito/ir/support/basic.py

```diff
@@ -426,7 +426,32 @@ def distance(self, other):
             else:
                 v = i - j
                 if v.is_Number and v.is_finite:
-                    return Vector(S.ImaginaryUnit)
+                    if i.is_Number and j.is_Number:
+                        return Vector(S.ImaginaryUnit)
+                    else:
+                        # If both i and j are not numbers,
+                        # there may be dimension-dependent dependencies
+                        # so we append the distance
+
+                        # For example:
+                        # Eq(u[0, y], 1)
+                        # Eq(u[1, 1], u[0, y+1])
+                        ret.append(S.Infinity)
+
+                # We are writing over an entire dimension
```

Comment:
you're breaking to a new line too soon. Use around 83~84 characters, then break line

Reply:
also simplify as "Writing (reading) over an entire dimension, reading (writing) from one point"